core_data_modules/cleaners/somali/demographic_cleaner.py

```diff
@@ -1,8 +1,86 @@
-from core_data_modules.cleaners import RegexUtils
+from core_data_modules.cleaners import RegexUtils, Codes, DigitCleaner
 from core_data_modules.cleaners.somali.demographic_patterns import Patterns
 
 
 class DemographicCleaner(object):
+    @staticmethod
+    def is_noise(text):
+        return RegexUtils.has_matches(text, Patterns.noise)
+
+    @staticmethod
+    def is_only_yes_no(text):
+        return RegexUtils.has_matches(text, Patterns.only_yes_no)
+
+    @staticmethod
+    def clean_gender(text):
+        return RegexUtils.clean_with_patterns(text, Patterns.genders)
+
+    @staticmethod
+    def clean_yes_no(text):
+        return RegexUtils.clean_with_patterns(text, Patterns.yes_no)
+
+    @staticmethod
+    def clean_urban_rural(text):
+        return RegexUtils.clean_with_patterns(text, Patterns.urban_rural)
+
     @staticmethod
     def clean_somalia_district(text):
         return RegexUtils.clean_with_patterns(text, Patterns.somalia_districts)
+    
+    @staticmethod
+    def clean_number_words(text):
+        """
+        Extracts the numbers in the given text that are expressed in words, and converts them to an integer.
+        
+        The strategy employed searches the given string for each number in somali.Patterns.numbers,
+        and returns the sum total of the numbers which matched at least once in the given text.
+
+        >>> DemographicCleaner.clean_number_words("lix iyo lawatan")
+        26
+        >>> DemographicCleaner.clean_number_words("afar shan lix")
+        15
+
+        :param text: Text to clean
+        :type text: str
+        :return: Extracted number
+        :rtype: int | Codes.NotCleaned
+        """
+        total = 0
+        found_match = False
+
+        for number, pattern in Patterns.numbers.items():
+            if RegexUtils.has_matches(text, pattern):
+                found_match = True
+                total += number
+
+        if found_match:
+            return total
+        else:
+            return Codes.NotCleaned
+
+    @classmethod
+    def clean_age(cls, text):
+        """
+        Extracts the number from the given text from either digits or words.
+
+        Returns Codes.NotCoded if age <= 10 or age >= 90.
+
+        >>> DemographicCleaner.clean_age("lix iyo lawatan")
+        26
+        >>> DemographicCleaner.clean_age(" 35.")
+        '35'
+
+        :param text:
+        :type text: str
+        :return:
+        :rtype: str | int # TODO: depends on which helper function is called
```

Comment:
I'm not sure I understand the need for return type polymorphism here. Maybe we can chat briefly about it?